[rocPRIM][Code Coverage][Cherry Pick] Increase code coverage for rocPRIM (#753)
ROCm/rocm-libraries: projects/rocprim/CHANGELOG.md

* Fixed device radix sort not returning the correct required temporary storage when a double buffer contains `nullptr`.
* Fixed constness of equality operators (`==` and `!=`) in `rocprim::key_value_pair`.

> * Fixed an issue for the comparison operators in `arg_index_iterator` and `texture_cache_iterator`, where `<` and `>` comparators were swapped.

Reviewer: ```suggestion
* Fixed an issue in `arg_index_iterator` and `texture_cache_iterator`, where the less than (`<`) and greater than (`>`) comparators were swapped.
```

> * Fixed an issue for the `rocprim::thread_reduce` not working correctly with a prefix value.

Reviewer: ```suggestion
* Fixed the issue in `rocprim::thread_reduce` where it wasn't working correctly with a prefix value.
```
